### PR TITLE
Add spacemouse SE3/SE2 device: schema, tracker, plugin, retargeters

### PR DESCRIPTION
**Pull request by rwiltz**
Ports Isaac Lab's Se2SpaceMouse/Se3SpaceMouse off the raw HID protocol onto the IsaacTeleop session API. Validated against IsaacLab's existing implementation and pyspacemouse (a third-party Python
  SpaceMouse SDK) to confirm protocol details.

  - spacemouse.fbs + (codegen'd) SpaceMouseTracker: raw translation/rotation axes + button state, no semantic mapping.
  - spacemouse plugin: standalone Linux HID reader (hidraw), self-discovers its device via HID_NAME sysfs matching against 4 known SpaceMouse product families (SpaceMouse Compact, Wireless, SpaceNavigator
  for Notebooks, and the 3Dconnexion Universal Receiver's combined-report layout), auto-launched by PluginManager.
  - SpaceMouseSource (IDeviceIOSource) exposing "spacemouse_translation", "spacemouse_rotation" (3-entry float arrays), and "spacemouse_buttons" (8-entry bitmap) -- reported as-is; retargeters own all
  mapping.
  - SpaceMouseToSe3RelRetargeter + SpaceMouseGripperRetargeter (translation/rotation -> EE delta, left-button toggle -> gripper) and SpaceMouseToSe2Retargeter (translation/rotation -> base velocity),
  matching the legacy Se3SpaceMouse/Se2SpaceMouse axis mapping and sensitivities -- including Se2's non-inverted omega_z, a faithful port of the legacy implementation's asymmetry with Se3's rotation
  mapping (documented in-code).
  - spacemouse_printer_example.py: live translation/rotation/button printer, mirroring the keyboard/gamepad examples.
 
  Fixes #(issue)

  ## Type of change

  - [ ] Bug fix (non-breaking change which fixes an issue)
  - [x] New feature (non-breaking change which adds functionality)
  - [ ] Breaking change (fix or feature that would cause existing functionality to change)
  - [ ] Documentation update
  
  ## Testing
 
  No physical SpaceMouse device was available for live testing. Verified structurally: OpenXR session creation, tracker registration, and plugin auto-launch all succeed; the plugin correctly reports "No
  SpaceMouse-family device found" and exits cleanly when no hardware is attached (the expected, correct failure mode). Full schema/source/retargeter unit-test suite (20 tests) passes with synthetic data
  covering translation/rotation mapping, single- and multi-button bitmaps, gripper toggle (including the reset-frame edge case), and inactive-device defaults.

  ## Checklist

  - [x] I have read and understood the [contribution guidelines](../CONTRIBUTING.md)
  - [x] I have run the linter and formatter with `SKIP=check-copyright-year pre-commit run --all-files`
  - [x] I have made corresponding changes to the documentation
  - [x] I have added tests that prove my fix/feature works (or explained why not)
  - [x] I have signed off all my commits (`git commit -s`) per the [DCO](../CONTRIBUTING.md#signing-your-work)

### CODE REVIEW

File: src/core/schema/fbs/spacemouse.fbs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+// All fields are always present whenever this table itself is present.
+table SpaceMouseOutput {
+    // Translation axis readings [x, y, z], normalized to [-1, 1]. Index i is the
+    // value of translation axis i, in the order reported by the device's HID report
+    // (see spacemouse_plugin.cpp).
+    translation: [float] (id: 0);
+
+    // Rotation axis readings [x, y, z], normalized to [-1, 1], same convention as
+    // translation.
+    rotation: [float] (id: 1);
```
Comment (aristarkhovNV) [inline src/core/schema/fbs/spacemouse.fbs:26]: can we use existing `Pose` schema type to represent the transform?

File: src/core/schema/fbs/spacemouse.fbs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    // Button indices currently held down (bit position in the device's button
+    // report byte), in no particular order.
+    pressed_buttons: [ushort] (id: 2);
+
+    // Whether the tracker has emitted at least one sample.
+    is_valid: bool (id: 3);
```
Comment (aristarkhovNV) [inline src/core/schema/fbs/spacemouse.fbs:33]: If the record exists then apparently there is at least one sample. The field does not seem to add much value by itself

File: src/plugins/spacemouse/spacemouse_plugin.cpp [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    pressed_buttons_.clear();
+}
+
+void SpaceMousePlugin::push_current_state()
+{
+    core::SpaceMouseOutputT out;
+    out.translation.assign(translation_.begin(), translation_.end());
+    out.rotation.assign(rotation_.begin(), rotation_.end());
+    out.pressed_buttons.assign(pressed_buttons_.begin(), pressed_buttons_.end());
+    out.is_valid = true;
```
Comment (aristarkhovNV) [inline src/plugins/spacemouse/spacemouse_plugin.cpp:190]: cannot seem to be other than `true`, consider removing

File: src/plugins/spacemouse/spacemouse_plugin.cpp [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+SpaceMousePlugin::SpaceMousePlugin(const std::string& device_path, const std::string& collection_id, bool combined_report)
+    : device_path_(device_path),
+      combined_report_(combined_report),
+      session_(std::make_shared<core::OpenXRSession>("SpaceMousePlugin", core::SchemaPusher::get_required_extensions())),
+      pusher_(session_->get_handles(),
+              core::SchemaPusherConfig{ .collection_id = collection_id,
+                                        .max_flatbuffer_size = kMaxFlatbufferSize,
+                                        .tensor_identifier = "spacemouse",
+                                        .localized_name = "SpaceMouse",
+                                        .app_name = "SpaceMousePlugin" })
```
Comment (aristarkhovNV) [inline src/plugins/spacemouse/spacemouse_plugin.cpp:60]: find a way to communicate the device name, especially when it is auto-discovered so it could be preserved in the recording as a channel metadata
